File: CSharp/basic/Trivia/GameRunner.cs
using System;

namespace Trivia
{
    public class GameRunner
    {

        private static bool notAWinner;

        public static void Main(String[] args)
        {
            Game aGame = new Game();

            aGame.Add("Chet");
            aGame.Add("Pat");
            aGame.Add("Sue");

            Random rand = new Random();

            do
            {

                aGame.Roll(rand.Next(5) + 1);

                if (rand.Next(9) == 7)
                {
                    notAWinner = aGame.WrongAnswer();
                }
                else
                {
                    notAWinner = aGame.WasCorrectlyAnswered();
                }



            } while (notAWinner);

        }


    }

}

File: scala/src/main/scala/com/adaptionsoft/games/trivia/runner/GameRunner.scala
package com.adaptionsoft.games.trivia.runner

import java.util.Random

import com.adaptionsoft.games.uglytrivia.Game

object GameRunner {
  private var notAWinner = false

  def main(args: Array[String]): Unit = {
    val rand = new Random
    playGame(rand)
  }

  def playGame(rand: Random): Unit = {
    val aGame = new Game
    aGame.add("Chet")
    aGame.add("Pat")
    aGame.add("Sue")
    do {
      aGame.roll(rand.nextInt(5) + 1)
      if (rand.nextInt(9) == 7) notAWinner = aGame.wrongAnswer
      else notAWinner = aGame.wasCorrectlyAnswered
    } while ( {
      notAWinner
    })
  }
}
